tests/scale/README.md

```diff
@@ -0,0 +1,398 @@
+  ready).
+- Record the time to ready in seconds in a csv file for each iteration.
+
+Total Resources Created:
+
+- 1 Gateway with 1 Listener
+- 1000 HTTPRoutes
+- 1 Service, Deployment, Pod
+
+This test takes around 7 hours to run, so I recommend running it on a VM, or overnight with the aid of
+[caffeinate](https://www.theapplegeek.co.uk/blog/caffeinate) for MAC users.
+
+Follow the steps below to run the test:
+
+- Run the test:
+
+  ```console
+  go test -v -tags scale -timeout 600m -run TestScale_HTTPRoutes -i 1000 -delay 2s
+  ```
+
+- [Collect and Record Metrics](#collecting-metrics).
+
+- Cleanup
+
+  Delete resources from cluster:
+
+  ```console
+  kubectl delete -Rf TestScale_HTTPRoutes
+  ```
+
+  Delete generated manifests:
+
+  ```console
+  rm -rf TestScale_HTTPRoutes
+  ```
+
+- Check for any errors or restarts after cleanup.
+- Check NGINX conf to make sure it looks correct.
+
+#### Scale Upstream Servers
+
+Test Goal: Measure how NGF performs as the number of Upstream Servers increases to the max of 648.
+
+Test Plan:
+
+- Deploy a single Gateway with 1 Listener and attach one HTTPRoute that references a single Service
+- Scale the deployment for that Service to 648 Pods (this is the limit that the upstream zone size allows)
+- Gateway, HTTPRoute, Service, and Deployment with 1 replica are created before scaling up to 648 replicas.
+
+Total Resources Created:
+
+- 1 Gateway with 1 Listener
+- 1 HTTPRoutes
+- 1 Service, 1 Deployment, 648 Pods
+
+For this test you must use a much bigger cluster in order to create 648 Pods. Use the following cluster details as a
+guide:
+
+- 12 n2d-standard-16
+- 192 vCPUs
+- 768 GB total memory
+- us-west2-b
+- 1.27.6-gke.1248000
+
+Follow the steps below to run the test:
+
+- Apply manifest
+
+  ```console
+  kubectl apply -f manifests/scale-upstreams.yaml
+  ```
+
+- Check the status of the Gateway and HTTPRoute to make sure everything is OK before scaling.
+
+  ```console
+  kubectl describe gateway gateway
+  kubectl describe httproute route
+  ```
+
+- Get the start time as a UNIX timestamp and record it in the
+  results [summary](/tests/scale/results/summary.md#upstream-servers):
+
+  ```console
+  date +%s
+  ```
+
+  This will be used in the metrics query.
+
+- Open a new terminal window and start the following loop:
+
+  ```console
+  for i in $(seq 1 150); do curl --resolve cafe.example.com:80:$NGF_IP http://cafe.example.com:80/; sleep 1; done >> requests.log
+  ```
+
+- Back in your original terminal, scale the backend app:
+
+  ```console
+  kubectl scale deploy backend --replicas 648
+  ```
+
+- Wait for all Pods to become available:
+
+  ```console
+  watch kubectl get deploy backend
+  ```
+
+- Check the NGINX config for 648 upstream servers:
+
+  ```console
+  k exec -it -n nginx-gateway $NGF_POD -c nginx -- nginx -T | grep -E "server (?:[0-9]{1,3}\.){3}[0-9]{1,3}:8080" | wc -l
```

Comment:
```suggestion
  kubectl exec -it -n nginx-gateway $NGF_POD -c nginx -- nginx -T | grep -E "server (?:[0-9]{1,3}\.){3}[0-9]{1,3}:8080" | wc -l
```

Reply:
whoops. I wish everyone had the `k` alias! Thanks for catching